app/questionnaire/router.py

```diff
@@ -246,6 +237,24 @@ def full_routing_path(self):
 
         return path
 
+    def _get_first_incomplete_location(self, path):
+        for location in path:
```

Comment:
This was wrapped in an `if path` before my move, but I dropped it because it should never be `None`?. Feel free to correct me if I'm mistaken.